src/selectors/per-thread/composed.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import { createSelector } from 'reselect';
+
+import { tabSlugs, type TabSlug } from '../../app-logic/tabs-handling';
+
+import type { Selector } from '../../types/store';
+import type { $ReturnType } from '../../types/utils';
+import type { Thread, JsTracerTable } from '../../types/profile';
+
+/**
+ * Infer the return type from the getStackAndSampleSelectorsPerThread function. This
+ * is done that so that the local type definition with `Selector<T>` is the canonical
+ * definition for the type of the selector.
+ */
+export type ComposedSelectorsPerThread = $ReturnType<
+  typeof getComposedSelectorsPerThread
+>;
+
+type NeededThreadSelectors = {
+  getThread: Selector<Thread>,
+  getIsNetworkChartEmptyInFullRange: Selector<boolean>,
+  getJsTracerTable: Selector<JsTracerTable | null>,
+};
+
+/**
+ * Create the selectors for a thread that have to do with either stacks or samples.
+ */
+export function getComposedSelectorsPerThread(
+  threadSelectors: NeededThreadSelectors
+): * {
+  /**
+   * Visible tabs are computed based on the current state of the profile. Some
+   * effort is made to not show a tab when there is no data available for it or
+   * when it's absurd.
+   */
+  const getUsefulTabs: Selector<$ReadOnlyArray<TabSlug>> = createSelector(
```

Comment:
Moving the function to the thread selectors, I decided to rename the selector, because I found that `getVisibleTabs` was less meaningful in this context.